StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/CustomerSheet/CustomerSheet.swift

```diff
@@ -150,15 +154,58 @@ internal enum CustomerSheetResult {
     var userCompletion: ((SheetResult) -> Void)?
 }
 
+enum TaskResult<T> {
+    case success(T)
+    case failure(Error)
+}
 extension CustomerSheet {
-    func loadPaymentMethods(completion: @escaping (Result<[STPPaymentMethod], Error>) -> Void) {
+    func loadPaymentMethodInfo(completion: @escaping (Result<([STPPaymentMethod], PersistablePaymentMethodOption?), Error>) -> Void) {
         Task {
-            do {
-                let paymentMethods = try await customerAdapter.fetchPaymentMethods()
-                let filteredPaymentMethods = paymentMethods.filter { $0.type == .card }
-                completion(.success(filteredPaymentMethods))
-            } catch {
-                completion(.failure(error))
+            await withTaskGroup(of: TaskResult<Any?>.self) { group in
```

Comment:
Instead of using task groups, could we reduce boilerplate by doing something like:
```
async let paymentMethods = try customerAdapter.fetchPaymentMethods()
async let selectedPaymentMethod = try self.customerAdapter.fetchSelectedPaymentMethodOption()
let (paymentMethods, selectedPaymentMethod) = try await (paymentMethods, selectedPaymentMethod)
// Do all the client-side processing and filtering
```
But I haven't used task groups before, so maybe I'm missing some benefits!

Reply:
Wow, this is really elegant. I think the use of TaskGroups is way overkill.  thanks!